webapp/src/userSettings.ts

```diff
@@ -11,4 +11,39 @@ class UserSettings {
     }
 }
 
-export {UserSettings}
+const keys = ['language', 'theme', 'lastBoardId', 'lastViewId', 'emoji-mart.last', 'emoji-mart.frequently']
+
+export function exportUserSettingsBlob(): string {
+    return window.btoa(exportUserSettings())
+}
+
+function exportUserSettings(): string {
+    const settings = Object.fromEntries(keys.map((key) => [key, localStorage.getItem(key)]))
+    settings.timestamp = `${Date.now()}`
+    return JSON.stringify(settings)
+}
+
+export function importUserSettingsBlob(blob: string): boolean {
+    return importUserSettings(window.atob(blob))
+}
+
+function importUserSettings(json: string): boolean {
+    const settings = parseUserSettings(json)
+    const timestamp = settings.timestamp
+    const lastTimestamp = localStorage.getItem('timestamp')
+    if (!timestamp || (lastTimestamp && Number(timestamp) <= Number(lastTimestamp))) {
+        return false
+    }
```

Comment:
if you save whenever you change one of the settings and clear the localstorage before loading any data, you shouldn't need to check for timestamps.

Reply:
Yeah, I think that would work. The need for the timestamp stems from the fact that this PR only saves the user settings on window close and so I wanted to ensure that the settings are not imported more than once per web view. If we later switch to export the settings dynamically as they change, we can drop the time stamp. We just have to remember to also reinstall the user script when the settings change.